presidio-analyzer/presidio_analyzer/predefined_recognizers/in_aadhaar_recognizer.py

```diff
@@ -0,0 +1,77 @@
+from stdnum import verhoeff
+from presidio_analyzer import Pattern, PatternRecognizer
+from typing import Optional, List, Tuple
+
+
+class InAadhaarRecognizer(PatternRecognizer):
+    """
+    Recognizes Indian UIDAI Person Identification Number ("AADHAAR").
+
+    Reference: https://en.wikipedia.org/wiki/Aadhaar
+    A 12 digit unique number that is issued to each individual by Government of India
+    :param patterns: List of patterns to be used by this recognizer
+    :param context: List of context words to increase confidence in detection
+    :param supported_language: Language this recognizer supports
+    :param supported_entity: The entity this recognizer can detect
+    :param replacement_pairs: List of tuples with potential replacement values
+    for different strings to be used during pattern matching.
+    This can allow a greater variety in input, for example by removing dashes or spaces.
+    """
+
+    PATTERNS = [
+        Pattern(
+            "AADHAAR (Low)",
+            r"\b[0-9]{12}\b",
+            0.05,
+        ),
+    ]
+
+    CONTEXT = [
+        "aadhaar",
+        "uidai",
+    ]
+
+    def __init__(
+        self,
+        patterns: Optional[List[Pattern]] = None,
+        context: Optional[List[str]] = None,
+        supported_language: str = "en",
+        supported_entity: str = "IN_AADHAAR",
+        replacement_pairs: Optional[List[Tuple[str, str]]] = None,
+    ):
+        self.replacement_pairs = (
+            replacement_pairs
+            if replacement_pairs
+            else [("-", ""), (" ", ""), (":", "")]
+        )
+        patterns = patterns if patterns else self.PATTERNS
+        context = context if context else self.CONTEXT
+        super().__init__(
+            supported_entity=supported_entity,
+            patterns=patterns,
+            context=context,
+            supported_language=supported_language,
+        )
+
+    def validate_result(self, pattern_text: str) -> bool:
+        """Determine absolute value based on calculation."""
+        sanitized_value = self.__sanitize_value(pattern_text, self.replacement_pairs)
+        return self.__check_aadhaar(sanitized_value)
+
+    @staticmethod
+    def __check_aadhaar(sanitized_value: str) -> bool:
```

Comment:
Could you please provide a reference to this validation logic? A quick search mentions the Verhoeff Algorithm. Is this the same?

Reply:
this is the most detailed article that can be used for referencing. While the knowledge is available in public domain , it is not acknowledged officially in the goverment documents. https://medium.com/@krs.sharath03/how-aadhar-number-is-generated-and-validated-3c3e7172e606